components/summary-list/src/SummaryList.tsx

```diff
@@ -11,14 +11,26 @@ export type SummaryListProps = SummaryListContainerProps & {
 
 const SummaryListComponent: FC<SummaryListProps> = ({
   classBlock = 'govuk-summary-list',
-  items,
+  items: _items,
   ...props
 }) => {
+  const items = _items.map(item => {
+    const hasActions = !!item.actions && (
+      !Array.isArray(item.actions) || item.actions.length > 0
+    );
+    return { item, hasActions };
+  });
+  const someActions = items.some(({ hasActions }) => hasActions);
+
   return (
     <SummaryListContainer {...props} classBlock={classBlock}>
-      {items.map((itemProps, i: number) => (
-        <SummaryListItem key={i}  {...itemProps} />
-      ))}
+      {items.map((itemProps, i: number) => {
+        // If some items have actions but this one doesn't, add the 'no-actions' modifier
+        // https://design-system.service.gov.uk/components/summary-list/#showing-rows-with-and-without-actions
+        const classModifiers = someActions && !itemProps.hasActions ? ['no-actions'] : [];
```

Comment:
`classModifiers` can technically already exist on the props, so we should seek to add an extra one.